includes/User.php

```diff
@@ -3124,15 +3133,25 @@ public function isAnon() {
 	}
 
 	/**
-	 * Get whether the user is the current performer
+	 * Get whether the instance represents the current performer
 	 *
-	 * Inspired by User::equals() and its usage in MW 1.25
 	 * @return Bool
 	 * @author Michał Roszka <michal@wikia-inc.com>
 	 */
 	public function isCurrent() {
+		// This is the first implementation. I still rely on $wgUser here
+		// and want to log all cases when mIsCurrent property is false
+		// for the current performer and fix it for future checks.
 		global $wgUser;
-		return $this->getName() === $wgUser->getName();
+		if ( !$this->mIsCurrent && $this->getName() === $wgUser->getName() ) {
```

Comment:
No, the name is a unique property, too. All the `User:Name` pages, for instance, need the name to be a conclusive identifier. MediaWiki 1.25 uses names for the same type of comparisons. I suspect the reason behind it is as follows:
* it is generally better to use string identifiers to identify a resource; means: integers should only be used for quantity and arithmetics;
* type-sensitive comparisons of numeric identifiers require the operands to be of the same type while they are integers when retrieved from the database by many ORMs and strings when provided as GET parameters. The name is a string and strings are strings. :-)
* type-insensitive comparison could have serious consequences for values like `false`, `0`, `'0'`, `''`, `null`, `0.1e2`, because PHP.